<launch>
	<!--include file="$(find sdu_frobitpro_factory)/launch/includes/simulation.launch"/-->
	<include file="$(find sdu_frobitpro_factory)/launch/includes/frobitpro.launch"/>

	<node pkg="rosbag" name="rosbag_record" type="record" args="-a"/>
			
	<group ns="HMI">
		<node pkg="wiimote" type="wiimote_node.py" name="wiimote" respawn="true"/>
		<!--node pkg="joy" type="joy_node" name="joy" output="screen"/-->
	</group>
	
	<group ns="fmDecisionMakers">
		<node pkg="sdu_armadillo4_2014-10-14" name="mission" type="mission.py" output="screen">
		    <param name="deadman_topic" value="/fmCommand/deadman" />
			<param name="cmd_vel_topic" value="/fmCommand/cmd_vel" />
			<param name="odom_topic" value="fmKonowledge/pose" />
			<param name="feedback_topic" value="/HMI/joy/set_feedback" />
			<param name="joy_topic" value="/HMI/joy" />
			<param name="scale" value="2" />
			<param name="status_topic" value="/fmData/status"/>		
			<param name="max_linear_velocity" value="0.5"/>
			<param name="max_angular_velocity" value="1"/>
			<param name="deadband" value="5"/>
			<param name="reduced_range" value="40"/>
			<param name="publish_frequency" value="20"/>
			<param name="odom_frame" value="world" />
			<param name="base_frame" value="base" />
			<param name="odom_topic" value="/fmKnowledge/pose"/> <!--param name="odom_topic" value="/base_pose_ground_truth"/-->
		</node>
	</group>	
			
	<group ns="fmExecutors">
		<node pkg="position_action_server" name="position_planner" type="positionGoalActionServer.py" output="screen">
			<param name="cmd_vel_topic" value="/fmCommand/cmd_vel" />
			<param name="odom_frame" value="world" />
			<param name="odom_topic" value="fmKnowledge/odom" />  <!-- only used if tf is not -->
			<param name="base_frame" value="base" />
			<param name="use_tf" value="True" />
			
			<param name="max_linear_velocity" value="0.5"/>
			<param name="max_angular_velocity" value="1.0"/>
			<param name="max_initial_angle_error" value="1"/>  <!-- robot turns in place if angle larger than this threshold -->
			
			<param name="max_distance_error" value="0.05"/>  <!-- goal is considered reached when distance to target is below this threshold -->
			<param name="max_angle_error" value="0.2"/>  <!-- retarder is active above this threshold -->
			<param name="retarder" value="0.1"/>  <!-- linear velocity is reduced by this factor when retarder is active -->
			
			<param name="lin_p" value="1.0"/>  <!-- PID parametres for linear velocity -->
			<param name="lin_i" value="0.0"/>
			<param name="lin_d" value="0.0"/>
			
			<param name="ang_p" value="1.0"/>  <!-- PID parametres for angular velocity -->
			<param name="ang_i" value="0.0"/>
			<param name="ang_d" value="0.0"/>
			
			<param name="int_max" value="0"/>  <!-- anti wind up protection common to linear and angular-->
		</node>	
	</group>
		
</launch>

